zulip/tests/test_api_functions.py

```diff
@@ -0,0 +1,71 @@
+import unittest
+import re
+import responses
+from zulip import Client
+
+from unittest import TestCase
+
+class TestClient(Client):
+    def __init__(self, config_file: str) -> None:
+        super().__init__(config_file=config_file)
+        self.responses = responses.RequestsMock()
+        self.responses.start()
+        self.add_responses()
+
+    def add_responses(self)  -> None:
+        # For call_on_each_event
+        self.responses.add(
+            responses.POST,
+            url="{}v1/register".format(self.base_url),
+            json={'queue_id': 10, 'last_event_id': -1, 'msg': '', 'result': 'success'},
+            status=200
+        )
+        self.responses.add(
+            responses.GET,
+            url="{}v1/events".format(self.base_url),
+            json={'result': 'success', 'msg': '', 'events': [{'id': 123}]},
+            status=200
+        )
+        # For add_reaction
+        self.responses.add(
+            responses.POST,
+            url=re.compile("{}v1/messages/([0-9]*)/reactions".format(self.base_url)),
+            json={'result': 'success', 'msg': ''},
+            status=200
+        )
```

Comment:
I think we want to do this individually in each test, for just the endpoints we expect that test to call.  

Additionally, I would configure it for the precise URL that the test is going to use, and assert what parameters were actually present in the HTTP request -- the goal here is to use this to verify that the API bindings made the precise request that we expected, not simulate a server.